### PR TITLE
[Feature] jdbc connector for postgresql

### LINKED ISSUE #7271: Support JDBC Connector
## Feature request

**Is your feature request related to a problem? Please describe.**

https://github.com/StarRocks/starrocks/issues/7268

**Describe the solution you'd like**

We want to access all RDS through JDBC Connector. Common MySQL and PostgreSQL connectors can be supported first.

- Provide user-defined MySQL and PostgreSQL External Catalog etc. 
- When the user is in the JDBC External catalog, they can query the tables of JDBC data sources without creating an external table .
- Using JDBC connector as the base class, you can query external data sources such as MySQL and PostgreSQL. Of course, you could use MySQLConnector or PostgreSQL Connector extends the JDBCConnector.

**Describe alternatives you've considered**

**Additional context**

If you're interested in this feature. you may
- provide JDBC connector design doc
- Currently, we support JDBC resource and [MySQL External Table.](https://docs.starrocks.com/zh-cn/main/using_starrocks/External_table#mysql-%E5%A4%96%E9%83%A8%E8%A1%A8) Refer to their implementation, like `JDBCResource` and `JDBCTable` related codes in FE.
- refer to hive connector to implement `ConnectorMetadata` etc
- design and implement JDBCClient in FE (TBD)

### PR DESCRIPTION
**Pull request by zombee0**
Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7271

## Problem Summary(Required) ：

followed #9695 

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 47 / 54 (87.04%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/jdbc/PostgresSchemaResolver.java|40|47|85.11%|[45, 46, 64, 65, 70, 71, 96]|
|:large_blue_circle:|com/starrocks/connector/jdbc/JDBCSchemaResolver.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/jdbc/JDBCMetadata.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 47 / 56 (83.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/jdbc/PostgresSchemaResolver.java|40|49|81.63%|[44, 45, 63, 64, 69, 70, 88, 89, 97]|
|:large_blue_circle:|com/starrocks/connector/jdbc/JDBCSchemaResolver.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/jdbc/JDBCMetadata.java|6|6|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/connector/jdbc/PostgresSchemaResolver.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import java.sql.Statement;
+import java.sql.Types;
+import java.util.Collection;
+
+import static java.lang.Math.max;
+
+public class PostgresSchemaResolver extends JDBCSchemaResolver {
+
+    @Override
+    public String configConnectUri(String uri, String dbName) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/jdbc/PostgresSchemaResolver.java:23]: IMO, the biggest difference between pg and mysql is the mapping relation in starrocks.  like 
| sr | mysql | pg |
| ---- | ---- | ---- |
| catalog | instance | database |
| database | database | schema |
| table | table | table |

I think you need check that jdbc_url must contain the database when creating the pg catalog. And add some comments at the main modification place.

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/connector/jdbc/PostgresSchemaResolver.java:23]: thanks for your review, to ensure the consistency of users' behavior, pg connector's uri doesn't contain database， just like mysql，but i give it the default database "postgres" by configConnectorUri，is it ok?
for the mapping relation，if we want support schema for postgres，we have to make some changes of the tablename parser to support tablename like catalog.database.schema.table or some other way just like concat schema and table as tablename

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/jdbc/PostgresSchemaResolver.java:23]: @DorianZheng PTAL

Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/connector/jdbc/PostgresSchemaResolver.java:23]: @zombee0 I agree with @stephen-shelby, when creating catalog for PG, user should provide database of PG. And in StarRocks, schema in PG is database in SR.

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/connector/jdbc/PostgresSchemaResolver.java:23]: ok, i got it. thank you!

**Comment by zombee0** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 34 / 45 (75.56%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/JDBCTable.java|0|4|00.00%|[111, 117, 118, 120]|
|:large_blue_circle:|com/starrocks/connector/jdbc/PostgresSchemaResolver.java|28|35|80.00%|[49, 50, 55, 56, 74, 75, 83]|
|:large_blue_circle:|com/starrocks/connector/jdbc/JDBCSchemaResolver.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/jdbc/JDBCMetadata.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 34 / 45 (75.56%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/JDBCTable.java|0|4|00.00%|[111, 117, 118, 120]|
|:large_blue_circle:|com/starrocks/connector/jdbc/PostgresSchemaResolver.java|28|35|80.00%|[49, 50, 55, 56, 74, 75, 83]|
|:large_blue_circle:|com/starrocks/connector/jdbc/JDBCSchemaResolver.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/jdbc/JDBCMetadata.java|4|4|100.00%|[]|

**Review by stephen-shelby — APPROVED** [review]

**Comment by zombee0** [comment]
run starrocks_admit_test

**Review by imay — APPROVED** [review]

**Pull request merged**